### PR TITLE
ES7: using ES_HOME env variable for embedded mode #7822

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/home/EsHomeDir.java  3
         

Complexity decreasing per file
==============================
+ modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4839703&bid=16070368)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/EsHomeDir.java  3
         

Complexity decreasing per file
==============================
+ modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4839703&bid=16070368)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/EsHomeDir.java  3
         

Complexity decreasing per file
==============================
+ modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4839703&bid=16070368)

File: modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java [hunk 1]
```diff
@@ -49,13 +49,14 @@ private String getPathOfDirAsString( final String value, final String defaultSub
...
-        return HomeDir.get().toFile().toPath().resolve( defaultSubPath ).toAbsolutePath().toString();
-    }
-
-    public String resolveElasticServerDir()
-    {
-        final String value = configuration.esServerDir();
-        return getPathOfDirAsString( value, "../elasticsearch" );
+        try
+        {
+            return Files.createDirectories( HomeDir.get().toFile().toPath().resolve( defaultSubPath ) ).toAbsolutePath().toString();
```
Comment (rymsha) [inline modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java]: why toFile().toPath() ?

Comment (anatol-sialitski) [reply modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java]: HomeDir.get() returns an instance of HomeDir, so we need to call toFole and toPath

**Pull request merged**